pkg/driver_cryptocell_310/psa_cryptocell_310/ecc_ed25519.c

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (C) 2023 TU Dresden
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     pkg_driver_cryptocell_310
+ * @{
+ *
+ * @file
+ * @brief       Glue code translating between PSA Crypto and the CryptoCell 310 EC 25519 curve APIs
+ *
+ * @author      Mikolai Gütschow <mikolai.guetschow@tu-dresden.de>
+ *
+ * @}
+ */
+
+#include "crys_ec_edw_api.h"
+#include "psa_error.h"
+#include "cryptocell_310_util.h"
+#include "string_utils.h"
+
+#define ENABLE_DEBUG    0
+#include "debug.h"
+
+extern CRYS_RND_State_t *rndState_ptr;
+
+psa_status_t psa_generate_ecc_ed25519_key_pair( uint8_t *priv_key_buffer,
+                                                uint8_t *pub_key_buffer,
+                                                size_t *priv_key_buffer_length,
+                                                size_t *pub_key_buffer_length)
+{
+    CRYS_ECEDW_TempBuff_t tmp;
+    CRYSError_t ret;
+
+    /* contains seed (private key), concatenated with public key */
+    uint8_t secret_key[CRYS_ECEDW_ORD_SIZE_IN_BYTES + CRYS_ECEDW_MOD_SIZE_IN_BYTES] = { 0x0 };
```

Comment:
One thing I just noticed: is there a reason why you write the key into this buffer first and then later memcpy it into priv_key_buffer, instead of directly writing it there?

Reply:
> But looking at it, I notice that the second `memcpy` is redundant, since it should already be written by the cryptocell API. Will remove that.

Done with b2349f4.